File: trunk/esper/src/test/java/net/esper/support/filter/SupportFilterServiceImpl.java
package net.esper.support.filter;

import net.esper.filter.FilterService;
import net.esper.filter.FilterHandle;
import net.esper.filter.FilterValueSet;
import net.esper.event.EventBean;
import net.esper.collection.Pair;

import java.util.List;
import java.util.LinkedList;
import java.util.Collection;

public class SupportFilterServiceImpl implements FilterService
{
    private List<Pair<FilterValueSet, FilterHandle>> added = new LinkedList<Pair<FilterValueSet, FilterHandle>>();
    private List<FilterHandle> removed = new LinkedList<FilterHandle>();

    public void evaluate(EventBean event, Collection<FilterHandle> matchList)
    {
        throw new UnsupportedOperationException();
    }

    public void add(FilterValueSet filterValueSet, FilterHandle callback)
    {
        added.add(new Pair<FilterValueSet, FilterHandle>(filterValueSet, callback));
    }

    public void remove(FilterHandle callback)
    {
        removed.add(callback);
    }

    public int getNumEventsEvaluated()
    {
        throw new UnsupportedOperationException();
    }

    public List<Pair<FilterValueSet, FilterHandle>> getAdded()
    {
        return added;
    }

    public List<FilterHandle> getRemoved()
    {
        return removed;
    }
}

File: NEsper.Tests/support/filter/SupportFilterServiceImpl.cs
///////////////////////////////////////////////////////////////////////////////////////
// Copyright (C) 2007 Esper Team. All rights reserved.                                /
// http://esper.codehaus.org                                                          /
// ---------------------------------------------------------------------------------- /
// The software in this package is published under the terms of the GPL license       /
// a copy of which has been included with this distribution in the license.txt file.  /
///////////////////////////////////////////////////////////////////////////////////////

using System;
using System.Collections.Generic;

using net.esper.collection;
using net.esper.compat;
using net.esper.events;
using net.esper.filter;

namespace net.esper.support.filter
{
	public class SupportFilterServiceImpl : FilterService
	{
	    private IList<Pair<FilterValueSet, FilterHandle>> added = new List<Pair<FilterValueSet, FilterHandle>>();
	    private IList<FilterHandle> removed = new List<FilterHandle>();

	    public void Add(FilterValueSet filterValueSet, FilterHandle callback)
	    {
	        added.Add(new Pair<FilterValueSet, FilterHandle>(filterValueSet, callback));
	    }

	    public void Remove(FilterHandle callback)
	    {
	        removed.Add(callback);
	    }

	    public IList<Pair<FilterValueSet, FilterHandle>> GetAdded()
	    {
	        return added;
	    }

	    public IList<FilterHandle> GetRemoved()
	    {
	        return removed;
	    }

        #region FilterService Members

        public long NumEventsEvaluated
        {
            get { throw new UnsupportedOperationException(); }
        }

        public void Evaluate(EventBean _event, IList<FilterHandle> matches)
        {
            throw new UnsupportedOperationException();
        }

        #endregion
    }
} // End of namespace
